### PR TITLE
Added datastore QueryParams

### PR DESCRIPTION
**Pull request by icellan**
refactored all list functions to use that instead of page / pageSize

### CODE REVIEW

File: datastore/models.go [hunk 1]
```diff
@@ -206,30 +206,38 @@ func (c *Client) GetModels(
...
 	fieldResults interface{},
 	timeout time.Duration,
 ) error {
 
+	if queryParams == nil {
+		// init a new empty object for the default queryParams
+		queryParams = &QueryParams{}
+	}
+	if queryParams.OrderByField == "id" {
+		queryParams.OrderByField = "_id" // use Mongo _id instead of default id field
```
Comment (mrz1836) [inline datastore/models.go:219]: Is this only for mongo?

**Review by mrz1836 — CHANGES_REQUESTED** [review]
Minor comment

File: datastore/models.go (hunk 1 above)
Comment (icellan) [reply datastore/models.go:219]: Yes!

Comment (icellan) [reply datastore/models.go:219]: Wrong location! :-/

**Review by mrz1836 — APPROVED** [review]

**Pull request merged**